### ISSUE TITLE
训练时候总是出现nan

### ISSUE THREAD

**Issue by tanghaojie1** [Ma-Dan/keras-yolo4#4]
loss和val loss总是出现nan

**Comment by chunleiml** [comment]
> loss和val loss总是出现nan

在哪里设置图像路径的？

**Comment by wenchao1993** [comment]
> loss和val loss总是出现nan
 我也是训练时loss和val出现nan, 然后发现把input图片的尺寸改成612x612，可以缓解这个问题。

**Comment by 6457tang** [comment]
> > loss和val loss总是出现nan
> > 我也是训练时loss和val出现nan, 然后发现把input图片的尺寸改成612x612，可以缓解这个问题。
我设置成612之后还是有这个问题

**Comment by tanghaojie1** [comment]
> > > loss和val loss总是出现nan
> > > 我也是训练时loss和val出现nan, 然后发现把input图片的尺寸改成612x612，可以缓解这个问题。
> > > 我设置成612之后还是有这个问题
他说的完善
感觉还是他的代码不太完善，是不是损失函数有问题

**Comment by Ma-Dan** [comment]
损失函数是有问题，也可能是权重初始化的参数有问题（那部分来源于yolov3），在加载预训练权重的前提下可以训练，这两部分都需要重写

**Comment by robisen1** [comment]
> 损失函数是有问题，也可能是权重初始化的参数有问题（那部分来源于yolov3），在加载预训练权重的前提下可以训练，这两部分都需要重写

Have you had a chance to fix the issues in train? thanks

**Comment by Kuuuo** [comment]
我是训练过程中mAP出现nan，这是什么问题呢？

Epoch 25/50
loss: [2.29048967][0][2.29048967][0]
1234/1234 [==============================] - 398s 323ms/step - loss: 3.6579.30782938][0][4.30782938][0]]]
Epoch end eval mAP on weight logs/000/ep025-loss3.658.h5
Epoch 25 mAP nan

**Comment by Mahmood-Hussain** [comment]
> 我是训练过程中mAP出现nan，这是什么问题呢？
> 
> Epoch 25/50
> loss: [2.29048967][0][2.29048967][0]
> 1234/1234 [==============================] - 398s 323ms/step - loss: 3.6579.30782938][0][4.30782938][0]]]
> Epoch end eval mAP on weight logs/000/ep025-loss3.658.h5
> Epoch 25 mAP nan

@Kuuuo have you been able to solve it?

**Comment by Kuuuo** [comment]
Maybe the learning rate is set too large




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "Ma-Dan/keras-yolo4"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年6月4日(星期五) 下午2:36
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [Ma-Dan/keras-yolo4] 训练时候总是出现nan (#4)





  
我是训练过程中mAP出现nan，这是什么问题呢？
 
Epoch 25/50
 loss: [2.29048967][0][2.29048967][0]
 1234/1234 [==============================] - 398s 323ms/step - loss: 3.6579.30782938][0][4.30782938][0]]]
 Epoch end eval mAP on weight logs/000/ep025-loss3.658.h5
 Epoch 25 mAP nan
  
@Kuuuo have you been able to solve it?

**Comment by Mahmood-Hussain** [comment]
> Maybe the learning rate is set too large
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "Ma-Dan/keras-yolo4" ***@***.***&gt;; 发送时间:&nbsp;2021年6月4日(星期五) 下午2:36 ***@***.***&gt;; ***@***.******@***.***&gt;; 主题:&nbsp;Re: [Ma-Dan/keras-yolo4] 训练时候总是出现nan (#4) 我是训练过程中mAP出现nan，这是什么问题呢？ Epoch 25/50 loss: [2.29048967][0][2.29048967][0] 1234/1234 [==============================] - 398s 323ms/step - loss: 3.6579.30782938][0][4.30782938][0]]] Epoch end eval mAP on weight logs/000/ep025-loss3.658.h5 Epoch 25 mAP nan @Kuuuo have you been able to solve it? — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub, or unsubscribe.

Thank You for your quick response.
by decreasing the learning rate did it solve your problem?

**Comment by Kuuuo** [comment]
yes




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "Ma-Dan/keras-yolo4"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年6月4日(星期五) 下午2:45
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [Ma-Dan/keras-yolo4] 训练时候总是出现nan (#4)





  
Maybe the learning rate is set too large
 …
 ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "Ma-Dan/keras-yolo4" @.&gt;; 发送时间:&nbsp;2021年6月4日(星期五) 下午2:36 @.&gt;; @.@.&gt;; 主题:&nbsp;Re: [Ma-Dan/keras-yolo4] 训练时候总是出现nan (#4) 我是训练过程中mAP出现nan，这是什么问题呢？ Epoch 25/50 loss: [2.29048967][0][2.29048967][0] 1234/1234 [==============================] - 398s 323ms/step - loss: 3.6579.30782938][0][4.30782938][0]]] Epoch end eval mAP on weight logs/000/ep025-loss3.658.h5 Epoch 25 mAP nan @Kuuuo have you been able to solve it? — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub, or unsubscribe.
  
Thank You for your quick response.
 by decreasing the learning rate did it solve your problem?

**Comment by Mahmood-Hussain** [comment]
> yes
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "Ma-Dan/keras-yolo4" ***@***.***&gt;; 发送时间:&nbsp;2021年6月4日(星期五) 下午2:45 ***@***.***&gt;; ***@***.******@***.***&gt;; 主题:&nbsp;Re: [Ma-Dan/keras-yolo4] 训练时候总是出现nan (#4) Maybe the learning rate is set too large … ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "Ma-Dan/keras-yolo4" @.&gt;; 发送时间:&nbsp;2021年6月4日(星期五) 下午2:36 @.&gt;; @.@.&gt;; 主题:&nbsp;Re: [Ma-Dan/keras-yolo4] 训练时候总是出现nan (#4) 我是训练过程中mAP出现nan，这是什么问题呢？ Epoch 25/50 loss: [2.29048967][0][2.29048967][0] 1234/1234 [==============================] - 398s 323ms/step - loss: 3.6579.30782938][0][4.30782938][0]]] Epoch end eval mAP on weight logs/000/ep025-loss3.658.h5 Epoch 25 mAP nan @Kuuuo have you been able to solve it? — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub, or unsubscribe. Thank You for your quick response. by decreasing the learning rate did it solve your problem? — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub, or unsubscribe.

Great! thanks

**Comment by Mahmood-Hussain** [comment]
@Kuuuo can you tell me what learning rate you set and what is anchor.txt file ?

**Comment by wangzhengzzz** [comment]
> 函数损失是有问题，也可能是权重初始化的参数有问题（那部分来自yolov3），在加载预训练重的基础可以训练，这部分权利都需要补充

您好！请问可以自己修改损失函数来解决loss为nan的问题吗？在哪里修改呢
